Update k8s tests and libraries to v1.24.0

aanm opened the pull request:
Signed-off-by: André Martins <andre@cilium.io>

```release-note
Add support for Kubernetes v1.24.0
```

*Note for reviewers*: This PR was tested against kernel 4.9 and k8s 1.24. The tests that are currently failing are known as they are also failing for other Kubernetes versions and thus not being k8s specific. Since it's currently not possible to update the net-next VM I'm proposing to change running the net-next w/ k8s 1.24 to running kernel 4.9 with k8s 1.24.

aanm committed:
Update k8s tests and libraries to v1.24.0

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
vagrant: update dev VM to 142

This VM contains the containerd container rumtime that is going to be
used with Kubernetes 1.24 as Kubernetes dropped support for docker.

Switched the container runtime to containerd since that is already
available in K8s 1.23 and we can start using it this version. If we keep
docker, the container images will have to be pulled by the docker container
runtime making the tests slower and eventually timeout.

Fixed the pre-pull of docker images while provisioing the VM

Signed-off-by: André Martins <andre@cilium.io>

nebril commented on jenkinsfiles/ginkgo-kernel.Jenkinsfile:
```diff
@@ -54,17 +54,17 @@ pipeline {
                         if [ "${ghprbCommentBody}" != "" ]; then
                             python3 ${TESTDIR}/get-gh-comment-info.py ''' + flags + ''' --retrieve="k8s_version" | \
-                            sed "s/^$/${JobK8sVersion:-1.23}/" | \
+                            sed "s/^$/${JobK8sVersion:-1.24}/" | \
```
I don't think any jenkins job uses the default value RN, we still need to configure new jobs with `1.24`.

aanm committed:
test: quarantine Checks E/W loadbalancing tests

These tests are currently failing for other Kubernetes versions and they
are now failing for these versions. Thus, we should quarantine them to
unblock the CI.

Signed-off-by: André Martins <andre@cilium.io>
